src/sentry/taskworker/client.py

```diff
@@ -163,7 +182,10 @@ def update_task(
         )
         try:
             with metrics.timer("taskworker.update_task.rpc"):
-                response = self._stub.SetTaskStatus(request)
+                if task_id not in self._task_id_to_stub_idx:
+                    metrics.incr("taskworker.client.task_id_not_in_client")
+                    return None
+                response = self._stubs[self._task_id_to_stub_idx[task_id]].SetTaskStatus(request)
```

Comment:
When do we remove from `_task_id_to_stub_idx`? It looks like it will grow indefinitely right now.  The task fetched via `has_next` that is returned on 198, should also be added to the `task_id_to_stub_idx` map so that updates can be done later.